### PR TITLE
Basic MIPS ELF support

### PR DESCRIPTION
**Pull request by jonas-schievink**

### CODE REVIEW

File: src/read/elf/file.rs [hunk 1]
```diff
@@ -116,6 +116,7 @@ where
             elf::EM_386 => Architecture::I386,
             elf::EM_X86_64 => Architecture::X86_64,
             elf::EM_MIPS => Architecture::Mips,
+            elf::EM_MIPS_RS3_LE => Architecture::Mips,
```
Comment (bjorn3) [inline src/read/elf/file.rs:119]: This one probably needs a new `Architecture` variant.

Comment (jonas-schievink) [reply src/read/elf/file.rs:119]: I've noticed that `llvm-objdump` displays ELFs of type `EM_MIPS_RS3_LE` as "file format ELF32-unknown", so maybe in practice it isn't used and should instead be removed? `mipsel` can presumably also be detected via the ELF's endianness.

Comment (philipc) [reply src/read/elf/file.rs:119]: How did you generate a file with `EM_MIPS_RS3_LE`?

Comment (jonas-schievink) [reply src/read/elf/file.rs:119]: With this crate – originally the `write` module used `EM_MIPS` for big-endian and `EM_MIPS_RS3_LE` for little-endian. Then I saw that the `llvm-objdump` didn't like that, and that `clang --target=mipsel-unknown-none` would also use `EM_MIPS`.

**Comment by philipc** [comment]
I'm not sure if you're looking for feedback on this, or how you're approaching this, but I think the ideal way to do initial support for this is to generate a .o file using llvm, and then implement whatever is needed so that the `objcopy` example in this crate can preserve all the information in the .o file.

File: src/read/elf/file.rs (hunk 1 above)
Comment (jonas-schievink) [reply src/read/elf/file.rs]: Removed this part

**Comment by jonas-schievink** [comment]
With stubbed-out relocation parsing, objcopy now works properly. The objdump output also makes sense. Let me know if you want me to properly implement relocations, but the current state should be enough for what I need.

**Review by philipc — APPROVED** [review]
Thanks!

**Pull request merged**